lib/msf/ui/debug.rb

```diff
@@ -0,0 +1,278 @@
+        #
+        # [06/19/2020 12:05:02] [i(0)] core: Trying to continue despite failed database creation: could not connect to server: Connection refused
+        # 	Is the server running on host "127.0.0.1" and accepting
+        # 	TCP/IP connections on port 5433?
+        #
+        # [05/15/2020 14:19:20] [e(0)] core: [-] Error while running command debug: can't modify frozen String
+        # Call stack:
+        # /Users/Shared/Relocated_Items/Security/rapid7/metasploit-framework/lib/msf/ui/debug.rb:33:in `get_all'
+        # /Users/Shared/Relocated_Items/Security/rapid7/metasploit-framework/lib/msf/ui/console/command_dispatcher/core.rb:318:in `cmd_debug'
+        # /Users/Shared/Relocated_Items/Security/rapid7/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:523:in `run_command'
+        # /Users/Shared/Relocated_Items/Security/rapid7/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:474:in `block in run_single'
+        # /Users/Shared/Relocated_Items/Security/rapid7/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:468:in `each'
+        # /Users/Shared/Relocated_Items/Security/rapid7/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:468:in `run_single'
+        # /Users/Shared/Relocated_Items/Security/rapid7/metasploit-framework/lib/rex/ui/text/shell.rb:158:in `run'
+        # /Users/Shared/Relocated_Items/Security/rapid7/metasploit-framework/lib/metasploit/framework/command/console.rb:48:in `start'
+        # /Users/Shared/Relocated_Items/Security/rapid7/metasploit-framework/lib/metasploit/framework/command/base.rb:82:in `start'
+        #
+        # [06/19/2020 11:51:44] [d(2)] core: Stager osx/armle/reverse_tcp and stage osx/x64/meterpreter have incompatible architectures: armle - x64
+        #
+        # [05/15/2020 14:23:55] [e(0)] core: [-] Error during IRB: undefined method `[]' for nil:NilClass
+        res = errors.scan(%r|\[\d{2}/\d{2}/\d{4} \d{2}:\d{2}:\d{2}\] \[e\(\d+\)\] (?:(?!\[\d{2}/\d{2}/\d{4} \d{2}:\d{2}:\d{2}\] \[[A-Za-z]\(\d+\)\]).)+|m)
+
+        if res.empty?
+          return build_section('Errors',
+                               'The following errors occurred before the issue occurred:',
+                               'The error log file was empty')
+        end
+
+        # Scan returns each error as a single item array
+        res.flatten!
+
+        errors_str = concat_str_array_from_last_idx(res, ERROR_TOTAL)
+        build_section('Errors',
+                      'The following errors occurred before the issue occurred:',
+                      errors_str)
+      rescue StandardError => e
+        section_build_error('Failed to extract Errors', e)
+
+      end
+
+      def self.logs
+
+        log_lines = File.readlines(File.join(Msf::Config.log_directory, 'framework.log'))
+
+        logs_str = concat_str_array_from_last_idx(log_lines, LOG_LINE_TOTAL)
+
+        build_section('Logs',
+                      'The following logs were recorded before the issue occurred:',
```

Comment:
As in putting messages on the log file to say "Hey, a new shell was opened here" and some identifying info?

Reply:
Sorry, I meant the time at once msfconsole was opened. We probably don't care about stale logs.

Reply:
The comment was in a similar vein to https://github.com/rapid7/metasploit-framework/pull/13430/files#r444918893 - were we probably don't care about 10 errors ago 🐛

Reply:
This is kind of what I was going for with the [optional arguments](https://github.com/adamgalway-r7/metasploit-framework/pull/1), as it would let people choose how long the log line total and error total are. However, you're also right that there's no real reason people would care about details spanning multiple console sessions, and manually guess working that with optional log line numbers would be a pain.

Reply:
Would say this is more of a seperate PR to improve logging, then debug can build on said improvements